src/app/context/EthereumWalletContext.tsx

```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { createContext, useContext, ReactNode } from "react";
+import { useAccount, useDisconnect } from "wagmi";
+import { injected } from "wagmi/connectors";
+
+interface EthereumWalletContextProps {
+  ethAddress: string | undefined;
+  connectEthWallet: () => void;
+  disconnectEthWallet: () => void;
+  isEthConnected: boolean;
+}
+
+const EthereumWalletContext = createContext<
+  EthereumWalletContextProps | undefined
+>(undefined);
+
+export const EthereumWalletProvider = ({
+  children,
+}: {
+  children: ReactNode;
+}) => {
+  const connector = injected({
+    target() {
+      return {
+        id: "windowProvider",
+        name: "Window Provider",
+        provider: window.ethereum,
+      };
+    },
+  });
+  const { address, isConnected } = useAccount();
+  console.log("connector", connector);
+  //   const { connect } = useConnect({
+  //     connector: connector, // You can specify supported chains here
+  //   });
+  const { disconnect } = useDisconnect();
+
+  const connectEthWallet = () => {

```

Comment:
any reason the body of this function is commented?

Reply:
removed